bevy_lint/src/lints/restriction/missing_reflect.rs

```diff
@@ -88,36 +84,27 @@ impl<'tcx> LateLintPass<'tcx> for MissingReflect {
     fn check_crate(&mut self, cx: &LateContext<'tcx>) {
         // Finds all types that implement `Reflect` in this crate.
         let reflected: Vec<TraitType> =
-            TraitType::from_local_crate(cx.tcx, &crate::paths::REFLECT).collect();
+            TraitType::from_local_crate(cx, &crate::paths::REFLECT).collect();
 
         // Finds all non-`Reflect` types that implement `Event` in this crate.
-        let events: Vec<TraitType> = TraitType::from_local_crate(cx.tcx, &crate::paths::EVENT)
+        let events: Vec<TraitType> = TraitType::from_local_crate(cx, &crate::paths::EVENT)
             .filter(|trait_type| !reflected.contains(trait_type))
             .collect();
 
         // Finds all non-`Reflect` types that implement `Component` and *not* `Event` in this
         // crate. Because events are also components, we need to deduplicate the two to avoid
         // emitting multiple diagnostics for the same type.
-        let components: Vec<TraitType> =
-            TraitType::from_local_crate(cx.tcx, &crate::paths::COMPONENT)
-                .filter(|trait_type| {
-                    !(reflected.contains(trait_type) || events.contains(trait_type))
-                })
-                .collect();
+        let components: Vec<TraitType> = TraitType::from_local_crate(cx, &crate::paths::COMPONENT)
+            .filter(|trait_type| !(reflected.contains(trait_type) || events.contains(trait_type)))
+            .collect();
 
         // Finds all non-`Reflect` types that implement `Resource` in this crate.
-        let resources: Vec<TraitType> =
-            TraitType::from_local_crate(cx.tcx, &crate::paths::RESOURCE)
-                .filter(|trait_type| !reflected.contains(trait_type))
-                .collect();
+        let resources: Vec<TraitType> = TraitType::from_local_crate(cx, &crate::paths::RESOURCE)
+            .filter(|trait_type| !reflected.contains(trait_type))
+            .collect();
+
+        let reflect_trait_def_ids = crate::paths::PARTIAL_REFLECT.get(cx);
```

Comment:
The old `get_trait_def_id()` method was removed because it didn't handle multiple traits with the same name. [`PathLookup::get()`](https://docs.rs/clippy_utils/0.1.89/clippy_utils/paths/struct.PathLookup.html#method.get) is a more general version that returns an array of `DefId`s, instead of a single `DefId`. It's also less expensive, because the results are cached!